src/api/telemetry/TelemetryAPI.js

```diff
@@ -250,6 +250,121 @@ export default class TelemetryAPI {
     return options;
   }
 
+  /**
+   * Canonicalizes options object by creating a standardized version with sorted keys
+   * and recursively processed values. This ensures consistent object representation
+   * regardless of property order.
+   *
+   * Used internally to generate consistent hashes for caching subscription options.
+   *
+   * @private
+   * @param {Object|Array|*} options The options object to canonicalize
+   * @returns {Object|Array|*} A canonicalized version of the input where object keys are sorted
+   */
+  #canonicalizeOptions(options) {
```

Comment:
Could this just be replaced by sorting the Object keys on line 306?

eg.

```
const sortedKeys = Object.keys(value).sort();
for (const key of sortedKeys) {
...
```

Reply:
Great call! Will do.